src/module_4/module_4_model-fitting.md

```diff
@@ -0,0 +1,1214 @@
+      <td>0.0</td>
+      <td>...</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>2.0</td>
+      <td>33.0</td>
+      <td>42.0</td>
+      <td>31.134053</td>
+      <td>30.0</td>
+      <td>30.0</td>
+      <td>24.27618</td>
+    </tr>
+    <tr>
+      <th>4</th>
+      <td>33826472919172</td>
+      <td>ricepastapulses</td>
+      <td>2808429314180</td>
+      <td>3537167515780</td>
+      <td>2020-10-06 10:37:05</td>
+      <td>2020-10-06 00:00:00</td>
+      <td>3</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>...</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>2.0</td>
+      <td>33.0</td>
+      <td>42.0</td>
+      <td>31.134053</td>
+      <td>30.0</td>
+      <td>30.0</td>
+      <td>24.27618</td>
+    </tr>
+  </tbody>
+</table>
+<p>5 rows × 27 columns</p>
+</div>
+
+
+
+
+```python
+min_products = 5
+order_size = df_feature_frame.groupby('order_id')['outcome'].sum()
+orders_of_min_size = order_size[order_size >= min_products].index
+df_filtered = df_feature_frame[df_feature_frame['order_id'].isin(orders_of_min_size)]
+```
+
+
+```python
+info_cols = ['variant_id', 'order_id', 'user_id', 'created_at', 'order_date']
+label_col = 'outcome'
+features_cols = [col for col in df_filtered.columns if col not in info_cols + [label_col]]
+
+categorical_cols = ['product_type', 'vendor']
+binary_cols = ['ordered_before', 'abandoned_before', 'active_snoozed', 'set_as_regular']
+numerical_cols = [col for col in features_cols if col not in categorical_cols + binary_cols]
+```
+
+
+```python
+df_filtered = df_filtered.sort_values('created_at')
+
+order_ids = df_filtered['order_id'].unique()
+
+total_ids = len(order_ids)
+train_end = int(total_ids * 0.7)
+val_end = train_end + int(total_ids * 0.15)
+
+train_order_ids = order_ids[:train_end]
+val_order_ids = order_ids[train_end:val_end]
+test_order_ids = order_ids[val_end:]
+
+train_df = df_filtered[df_filtered['order_id'].isin(train_order_ids)]
+val_df = df_feature_frame[df_feature_frame['order_id'].isin(val_order_ids)]
+test_df = df_feature_frame[df_feature_frame['order_id'].isin(test_order_ids)]
+
+print(f"Train set size: {len(train_df)}")
+print(f"Validation set size: {len(val_df)}")
+print(f"Test set size: {len(test_df)}")
+```
+
+    Train set size: 1446691
+    Validation set size: 347233
+    Test set size: 370029
+
+
+
+```python
+def feature_label_split(df, label_col):
+    return df.drop(label_col, axis=1), df[label_col]
+
+X_train, y_train = feature_label_split(train_df, label_col)
+X_val, y_val = feature_label_split(val_df, label_col)
+X_test, y_test = feature_label_split(test_df, label_col)
+```
+
+
+```python
+"""""Función para crear los gráficos del performance de nuestro modelo"""
+def plot_metrics(model_name, y_pred, y_test, figsize=(10, 7)):
+    precision, recall, _ = precision_recall_curve(y_test, y_pred)
+    pr_auc = auc(recall, precision)
+    fpr, tpr, _ = roc_curve(y_test, y_pred)
```

Comment:
You are using y_test instead of y_eval. The idea of y_test is that you only touch it at the END when you have chose the final model and you want to know the final performance